docs/available-filters/envelope-encryption/envelope-encryption.adoc

```diff
@@ -0,0 +1,210 @@
+:kms-api-javadoc: https://javadoc.io/doc/io.kroxylicious/kroxylicious-kms/latest
+:encryption-api-javadoc: https://javadoc.io/doc/io.kroxylicious/kroxylicious-encryption
+:design-doc: https://github.com/kroxylicious/kroxylicious/blob/main/kroxylicious-filters/kroxylicious-encryption/doc/design.adoc
+
+= Envelope Encryption
+
+== What is it?
+
+A filter that provides a transparent https://kroxylicious.io/use-cases/[encryption-at-rest solution] for Apache Kafka.
+
+At a high level, the filter works in the following way.  First, let's consider the produce side:
+
+1. The filter intercepts produce requests sent from producing Kafka Clients.
+2. The filter disassembles the produce request.
+3. Each record within it is encrypted.
+4. The produce request is reassembled, replacing the original records with their encrypted counterparts.
+5. The filter forwards the modified produce request onto the Kafka Broker.
+6. The broker handles the records in the normal way writing them to the transaction log.  The broker has no knowledge
```

Comment:
The `transaction log` term is not used in Kafka, and might be confused with kafka's transactional log. Maybe use `data log`?

Reply:
I meant https://kafka.apache.org/documentation/#log. My intent was to emphasise to the reader that the broker has no knowledge of the fact that the record is encrypted.